### PR TITLE
[STORM-3003] Adding Assignment caching to Nimbus

### PR DESCRIPTION
**Pull request by kishorvpatil**
Since nimbus ( scheduler generates assignments) it can cache it instead of polling for it from ZK or other state manager. This would improve scheduling iteration time, as well as all UI pages that require assignment information.

The need for this improvement felt when we noticed this is larger clusters where ZK continues to be bottleneck.

### CODE REVIEW

**Comment by d2r** [comment]
The CI test error was in the nimbus test, `test-check-authorization-getSupervisorPageInfo`, and seems unrelated. @kishorvpatil does it look that way to you?

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 1]
```diff
@@ -1944,6 +1957,7 @@ private void mkAssignments(String scratchTopoId) throws Exception {
                 } else {
                     LOG.info("Setting new assignment for topology id {}: {}", topoId, assignment);
                     state.setAssignment(topoId, assignment);
+                    assignmentsCache.get().put(topoId, assignment);
```
[the remark this answers is not in the document]
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: This was addressed.

**Comment by kishorvpatil** [comment]
@d2r Thank you for the review. I have addressed the comments. The failure seems unrelated to my changes - the test passes on local environment.

**Comment by HeartSaVioR** [comment]
@kishorvpatil 
Could we hold on this for #2433 since #2433 is making huge change on relevant part? It would be really great if you have time to take a look at #2433 and check the conflict between this and #2433. 

Btw, regardless of #2433 I think this change can go through 1.x-branch, though separate patch is needed.

**Comment by kishorvpatil** [comment]
@HeartSaVioR  Sure. Let me review #2433. In the meantime, I will create patch for 1.x-branch.

**Comment by HeartSaVioR** [comment]
@kishorvpatil 
Now #2433 is merged in. Could you rebase and check this patch is still valid?

**Comment by kishorvpatil** [comment]
@HeartSaVioR , I think changes in #2433 do take care of caching assignments within `InMemoryAssignmentBackend`, so the changes proposed in #2603 are no longer needed. I am closing this PR

**Pull request closed without merging**